xxxx-release-and-versioning.md

```diff
@@ -0,0 +1,87 @@
+| 1.1.0   | 2017-06-20   | Not supported, not latest `1.1`            |
+| 1.0.2   | 2018-02-01   | Not supported, `1.0` discontinued (hotfix) |
+| 1.0.1   | 2017-08-01   | Not supported, `1.0` discontinued          |
+| 1.0.0   | 2016-08-01   | Not supported, `1.0` discontinued          |
+
+In the figure above, version `1.0.2` is released as a security fix outside of the support period. This release is considered a hotfix release. However, even though version `1.0.2` is released after the supported version `1.1.1`, the version `1.0.2` won't be guarantee to receive any further updates. Provider is encouraged to upgrade to latest supported version.
+
+### Development Release
+
+A development release is a release that provides bleeding edge features that may still be unstable, as the branch is only required to pass unit testing and integration testing. Usage of development release in a production environment is highly discouraged, and only recommended for testing and evaluation purpose.
+
+| Test subject        | Required |
+| ------------------- | -------- |
+| Linting             | **Yes**  |
+| Unit testing        | **Yes**  |
+| Integration testing | **Yes**  |
+| End to end testing  | No       |
+| Migration testing   | No       |
+
+The development release will be provided as a binary release and a Docker image based on Alpine Linux. The Docker image will be tagged with commit ID and will have a `:dev` tag to refer to the latest development release.
+
+### Branching Model
+
+The branching model will utilize a `master` branch, a stable version branch and a feature branch. To demostrate:
+
+```
+feature2    -o---------o----------o
+                                   \
+                                    \
+feature1    -o                       \
+              \                       \
+               v                       v
+master      ----o-----------------------o-------------------------------->
+                             \                           \
+                              v                           \
+stable-1.0                     o-----------o----o----------^------------->
+                            v1.0.0           v1.0.1         \
+                                                             v
+stable-1.1                                                    o---------->
+                                                            v1.1.0
+```
+
+#### Feature Branches
+
+Feature branches, such as `feature2` and `feature1`, are development branches that implements new feature or change existing functionality. These branches may be long-running and should be merged to `master` only when the feature is considered stable enough for public consumption. Feature branches should regularly sync up with master to avoid too much code diversion. Feature branches may have its own public testing environment for each branch with its data copied from staging on each deploy.
```

Comment:
@sirn @T-Dnzt Should we also mandate that feature branches are prefixed with `feature-`?

Also do we prefix our project-management or GitHub issue number in the branch?

Reply:
I and @T-Dnzt actually discussed about this tl;dr it's weird calling code cleanup a feature, and if we have multiple prefixes everyone would stop bothering about it at one point and just use feature/ for everything, which essentially the same as having no prefix.